mp/src/game/shared/momentum/util/engine_patch.cpp

```diff
@@ -0,0 +1,260 @@
+//==============================
+#ifdef _WIN32
+CEnginePatch g_EnginePatches[] =
+{
+    // Prevent the culling of skyboxes at high FOVs
+    // https://github.com/VSES/SourceEngine2007/blob/master/se2007/engine/gl_warp.cpp#L315
+    // TODO: Use a value derived from FOV instead
+    {
+        "SkyboxCulling",
+        "\xF3\x0F\x59\x15\x00\x00\x00\x00\xF3\x0F\x58\xC1\xF3\x0F\x10\x0D",
+        "xxxx????xxxxxxxx",
+        16,
+        PATCH_REFERENCE,
+        -1.0f
+    },
+    // Example patch: Trigger "Map has too many brushes" error at 16384 brushes instead of 8192
+    //{
+    //    "BrushLimit",
+    //    "\xC1\xEF\x03\x81\xFF\x00\x00\x00\x00",
+    //    "xxxxx????",
+    //    5,
+    //    PATCH_IMMEDIATE,
+    //    16384
+    //},
+    // The same patch as above but using a pure hex value
+    //{
+    //    "BrushLimitHex",
+    //    "\xC1\xEF\x03\x81\xFF\x00\x00\x00\x00",
+    //    "xxxxx????",
+    //    5,
+    //    PATCH_IMMEDIATE,
+    //    "\x00\x40\x00\x00"
+    //}
+};
+#elif __linux__
+CEnginePatch g_EnginePatches[] =
+{
+    // Prevent the culling of skyboxes at high FOVs
+    // https://github.com/VSES/SourceEngine2007/blob/master/se2007/engine/gl_warp.cpp#L315
+    // TODO: Use a value derived from FOV instead
+    {
+        "SkyboxCulling",
+        "\xF3\x0F\x59\x0D\x00\x00\x00\x00\xF3\x0F\x58\xC2\xF3\x0F\x58\xC1\xF3\x0F\x10\x0D",
+        "xxxx????xxxxxxxxxxxx",
+        20,
+        PATCH_REFERENCE,
+        -1.0f
+    },
+    // Example patch: Trigger "Map has too many brushes" error at 16384 brushes instead of 8192
+    //{
+    //    "BrushLimit",
+    //    "\xBE\x00\x00\x00\x00\xF7\xE6\x89\xD6\xC1\xEE\x03\x81\xFE",
+    //    "x????xxxxxxxxx",
+    //    14,
+    //    PATCH_IMMEDIATE,
+    //    16384
+    //},
+    // The same patch as above but using a pure hex value
+    //{
+    //    "BrushLimitHex",
+    //    "\xBE\x00\x00\x00\x00\xF7\xE6\x89\xD6\xC1\xEE\x03\x81\xFE",
+    //    "x????xxxxxxxxx",
+    //    14,
+    //    PATCH_IMMEDIATE,
+    //    "\x00\x40\x00\x00"
+    //}
+};
+#endif //_WIN32
+
+void CEnginePatch::ApplyAll()
+{
+#if !defined (OSX) // No OSX patches
+    int err;
+
+    for (int i = 0; i < sizeof(g_EnginePatches) / sizeof(*g_EnginePatches); i++)
+    {
+        err = g_EnginePatches[i].ApplyPatch();
+        switch (err)
+        {
+        case PATCH_SUCCESS:
+            DevLog("Engine patch \"%s\" applied successfully\n", g_EnginePatches[i].GetName());
+            break;
+        case PATCH_MEMPROTECT_FAIL:
+            Warning("Engine patch \"%s\" FAILED: Could not override memory protection\n", g_EnginePatches[i].GetName());
+            break;
+        case PATCH_INVALID_SIGNATURE:
+            Warning("Engine patch \"%s\" FAILED: Could not find signature\n", g_EnginePatches[i].GetName());
+            break;
+        case PATCH_NO_VALUE:
+            Warning("Engine patch \"%s\" FAILED: No value provided\n", g_EnginePatches[i].GetName());
+        }
```

Comment:
This can be done inside of `ApplyPatch`, which means `GetName` will not need to be exposed publicly anymore.